### ISSUE TITLE
[SYCL] Fortran with YAKL_SYCL_FLAGS

### ISSUE THREAD

**Issue by abagusetty** [mrnorman/YAKL#31]
Intel's Fortran compiler `ifx` seems to take `YAKL_SYCL_FLAGS` which might not be needed. Is there a way to shuffle the CMakeLists to pass `YAKL_SYCL_FLAGS` just to CXX and not to C or Fortran.

Build: `-DYAKL_SYCL_FLAGS="-O0 -g -sycl-std=2020 --intel -fsycl -fsycl-unnamed-lambda"`

```
cd /gpfs/jlse-fs0/users/abagusetty/projects/YAKL/unit/build/machines/jlse/Fortran_Link && /soft/packaging/spack-builds/linux-opensuse_leap15-x86_64/gcc-10.2.0/cmake-3.20.2-v26d3stmznkvg5wd5otjbacxkbhjtnu4/bin/cmake -E cmake_link_script CMakeFiles/Fortran_Link.dir/link.txt --verbose=1
/PATHTO/modulefiles/oneapi/bin-ifx/ifx CMakeFiles/Fortran_Link.dir/simple.cpp.o CMakeFiles/Fortran_Link.dir/wrapper.F90.o -o Fortran_Link  ../yakl/libyakl.a -O0 -g -sycl-std=2020 --intel -fsycl -fsycl-unnamed-lambda -lirng -lirng -lstdc++ 
ifx: command line warning #10006: ignoring unknown option '-sycl-std=2020'
ifx: command line warning #10006: ignoring unknown option '-fintel'
ifx: command line warning #10006: ignoring unknown option '-fsycl-unnamed-lambda'
ifx: command line warning #10006: ignoring unknown option '-sycl-std=2020'
ifx: command line warning #10006: ignoring unknown option '-fintel'
ifx: command line warning #10006: ignoring unknown option '-fsycl-unnamed-lambda'
ifx: command line warning #10006: ignoring unknown option '-sycl-std=2020'
ifx: command line warning #10006: ignoring unknown option '-fintel'
ifx: command line warning #10006: ignoring unknown option '-fsycl-unnamed-lambda'
ld: cannot find -lomptarget
ifx: error #10401: error running 'link for host dependence gathering'
make[2]: *** [Fortran_Link/CMakeFiles/Fortran_Link.dir/build.make:113: Fortran_Link/Fortran_Link] Error 1
```

**Comment by mrnorman** [comment]
Thanks for bringing this up.

I'm not sure why this is happening. Maybe CMake itself is adding these flags? Either way, there are a number of compilers that fail that unit test at the moment.

Isaac and I have been booted off of jlse, and there seem to be problems getting our accounts reinstated. 

Should I take this as a hint that Intel has implemented is_memcpyable ?

**Comment by abagusetty** [comment]
I was able to hack this by passing the SYCL flags explicitly `-DCMAKE_CXX_COMPILER` 

There has been a account migration on JLSE and you must have completed those steps. To get access to SDK and hardware queues, ORNL needs to sign off on the TOU which you might have received the email.

Intel has implemented parts of `is_memcpyable` but not yet released. I was testing all the unit tests, cleanup, bug fixes with the hack of using `functorBuffer`. Will create a PR (for this housecleaning) and also an active issue (to replace with `is_memcpyable`)

**Comment by mrnorman** [comment]
That sounds great, thanks! I recently looked into the functorBuffer thing due to issues with std::mutex in the Array class destructors. It didn't occur to me that the same approach would likely get us up and running with SYCL.

**Comment by abagusetty** [comment]
Addressed here in #34. Closing
